### PR TITLE
# Blog app - Add forms

### PR DESCRIPTION
**Pull request by Marcraphael12**
In this PR, we are adding a form to our app:
### Project requirements
-  Create a method called `current_user` in your ApplicationController to make current user data available to all controllers.
    -  It will return the first user from the database.
    -  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
- Create forms to perform the following functions:
    - Creates a Post on behalf of the `current_user` (use the method that you created in your ApplicationController).
    - Create a comment on behalf of the `current_user` (use the method that you created in your ApplicationController).
- Allow Users to add likes to Posts.

### CODE REVIEW

**Review by Lucas-Masaba — CHANGES_REQUESTED** [review]
# Hello @Marcraphael12  👋 , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request ⚠️ 
- [ ] You do not have stylelint checks on your pull request. Please make sure all linter checks are added so as to maintain the code standard.
image


Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @Lucas-Masaba in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Comment by Marcraphael12** [comment]
it's saturday already... and almost midnigth😩

File: app/models/user.rb [hunk 1]
```diff
@@ -7,9 +7,9 @@ class User < ApplicationRecord
 
   # validating that the `Name` attribute of the `User`
   # model is present, meaning it cannot be blank or nil.
-  validates :Name, presence: true
+  validates :name, presence: true
 
-  validates :PostCounter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
+  validates :post_ounter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```
Comment (Munsa1) [inline app/models/user.rb:12]: - [ ] You have a typo in your models causing an error preventing the creation of a User, therefore this makes it impossible to access the form and check whethera user is able to create a `post`, a `comment` and a like. Kindly fix this error.
Screenshot
image
- [ ] If you check the  screenshot above a post can be created, as well as a comment, only creating the user is throwing an error. Kindly fix this and ensure the forms are working as expected for posts and comments as well as adding likes to posts.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -15,4 +15,26 @@ def show
...
+      redirect_to user_path(id: @post.author_id)
+    else
+      render :new, alert: 'An error has occurred while creating the post'
+    end
+  end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
 end
```
Comment (Munsa1) [inline app/controllers/posts_controller.rb:40]: - [ ] (optional) Consider making the strong paramaters private for security reasons. Private methods are not meant to be called directly from outside the class. Making strong parameters private enforces encapsulation, ensuring that this part of your code is not exposed to external manipulation and adheres to the principle of information hiding.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      redirect_to user_post_path(user_id: @post.author_id, id: @post.id)
+    else
+      render :new, alert: 'An error has occurred while creating the comment'
+    end
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
+end
```
Comment (Munsa1) [inline app/controllers/comments_controller.rb:22]: - [ ] (optional) Consider making the strong paramaters private for security reasons. Private methods are not meant to be called directly from outside the class. Making strong parameters private enforces encapsulation, ensuring that this part of your code is not exposed to external manipulation and adheres to the principle of information hiding.

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @Marcraphael12, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Created Form Controllers :heavy_check_mark: 
Created Views For Form Controllers :heavy_check_mark: 
Descriptive ReadMe File :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
### Due to insufficient files changed

- [ ] Kindly remove the following link from your table of contents in your readme file as it points to a part that has been removed making it broken
image

- [ ] Kindly edit your MIT license agreement for a more descriptive readme file
image

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/comments_controller.rb (hunk 3 above)
Comment (Marcraphael12) [reply app/controllers/comments_controller.rb:22]: i'll apply that

File: app/controllers/posts_controller.rb (hunk 2 above)
Comment (Marcraphael12) [reply app/controllers/posts_controller.rb:40]: i'll apply that

File: app/models/user.rb (hunk 1 above)
Comment (Marcraphael12) [reply app/models/user.rb:12]: done

**Review by prolajumokeoni — APPROVED** [review]
Hi @Marcraphael12 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
  - Updated  Model files✅ 
   

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Marcraphael12** [comment]
🙏🏾😭😭😭

**Pull request merged**